### PR TITLE
Add pmd and findbugs.

### PR DESCRIPTION
**Pull request by ViddeM**
Will probably keep the branch to add checkstyle aswell.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -51,6 +51,29 @@ project(":desktop") {
 
 project(":core") {
     apply plugin: "java"
+    apply plugin: 'pmd'
+    apply plugin: 'findbugs'
+
+    findbugs{
+        ignoreFailures=true
```
Comment (Portals) [inline build.gradle]: This should be false. Why is this true? Travis should fail if there is rule validation

File: build.gradle [hunk 2]
```diff
@@ -51,6 +51,29 @@ project(":desktop") {
...
+        }
+    }
+    tasks.withType(Pmd){
+        reports{
+            xml.enabled=true
+            html.enabled=true
+        }
+    }
+    pmd {
+        ignoreFailures = true
```
Comment (Portals) [inline build.gradle]: Same as above

Comment (ViddeM) [reply build.gradle]: It was to ensure the build continuing even if there were warnings during setup, it's now fixed.

**Comment by ViddeM** [comment]
Also fixes some amount of bugs found with findbugs.

File: .travis.yml [hunk 3]
```diff
@@ -8,11 +8,4 @@ notifications:
...
         rooms:
-          digit1819:MI9cZ2mjZzXjEM42jwg4hJph
-
-    slack:
-        on_success: always
-        on_failure: always
-        
-        rooms:
-            digit1819:SG8nfPsbykYZdE3tR07XUUdy
+          digit1819:MI9cZ2mjZzXjEM42jwg4hJph
```
Comment (Portals) [inline .travis.yml]: Should this be a secret?

Comment (ViddeM) [reply .travis.yml]: It can be if we want to, if you want to generate a new one and encrypt it. Didn't feel it really mattered that much.

Comment (ViddeM) [reply .travis.yml]: Altough should be noted that this is the same as develop, I only removed the PM to me, getting double messages was starting to get annoying.

**Pull request merged**